packages/app-store/googlecalendar/api/webhook.ts

```diff
@@ -1,48 +1,140 @@
 import { defaultHandler } from "@calcom/lib/server/defaultHandler";
 import { defaultResponder } from "@calcom/lib/server/defaultResponder";
 import { SelectedCalendarRepository } from "@calcom/lib/server/repository/selectedCalendar";
+import type { SelectedCalendar } from "@calcom/prisma/client";
+import type { CredentialPayload } from "@calcom/types/Credential";
 
 import { getCalendar } from "../../_utils/getCalendar";
 
-async function postHandler(req: NextApiRequest) {
-  if (req.headers["x-goog-channel-token"] !== process.env.GOOGLE_WEBHOOK_TOKEN) {
-    throw new HttpError({ statusCode: 403, message: "Invalid API key" });
+const log = logger.getSubLogger({ prefix: ["googlecalendar", "api", "webhook"] });
+
+async function getCredentialForCalendarService(selectedCalendarMatchingGoogleChannelId: {
+  credential: CredentialPayload | null;
+  delegationCredentialId: string | null;
+  userId: number;
+  googleChannelId: string;
+}) {
+  if (selectedCalendarMatchingGoogleChannelId.credential) {
+    return buildNonDelegationCredential(selectedCalendarMatchingGoogleChannelId.credential);
+  } else if (selectedCalendarMatchingGoogleChannelId.delegationCredentialId) {
+    return findUniqueDelegationCalendarCredential({
+      userId: selectedCalendarMatchingGoogleChannelId.userId,
+      delegationCredentialId: selectedCalendarMatchingGoogleChannelId.delegationCredentialId,
+    });
+  } else {
+    throw new HttpError({
+      statusCode: 200,
+      message: `No credential or DelegationCredential found for selected calendar for googleChannelId: ${selectedCalendarMatchingGoogleChannelId.googleChannelId}`,
+    });
   }
-  if (typeof req.headers["x-goog-channel-id"] !== "string") {
-    throw new HttpError({ statusCode: 403, message: "Missing Channel ID" });
+}
+
+function getSelectedCalendarsForGoogleChannel(selectedCalendarMatchingGoogleChannelId: {
+  credential: { selectedCalendars: SelectedCalendar[] } | null;
+  delegationCredential: { selectedCalendars: SelectedCalendar[] } | null;
+  userId: number;
+  googleChannelId: string;
+}) {
+  if (selectedCalendarMatchingGoogleChannelId.credential) {
+    // Regular credential case -> There could be multiple selected calendars for the same user because of different eventTypes
+    return selectedCalendarMatchingGoogleChannelId.credential.selectedCalendars;
+  } else if (selectedCalendarMatchingGoogleChannelId.delegationCredential) {
+    // Because same delegationCredential is used by all members of the same organization
+    const selectedCalendarsForAllTheMembersOfTheOrganization =
+      selectedCalendarMatchingGoogleChannelId.delegationCredential.selectedCalendars;
+
+    // There could be multiple selected calendars for the same user because of different eventTypes
+    return selectedCalendarsForAllTheMembersOfTheOrganization.filter(
+      (selectedCalendar) => selectedCalendar.userId === selectedCalendarMatchingGoogleChannelId.userId
```

Comment:
Because same delegationCredential is used by all members of the same organization, we need to identify the SelectedCalendar with the correct userId match.